mlflow/diviner.py

```diff
@@ -208,47 +277,66 @@ def load_model(model_uri, dst_path=None):
                       `Referencing Artifacts <https://www.mlflow.org/docs/latest/tracking.html#
                       artifact-locations>`_.
     :param dst_path: The local filesystem path to which to download the model artifact.
-                     This directory must already exist. If unspecified, a local output
+                     This directory must already exist if provided. If unspecified, a local output
                      path will be created.
 
     :return: A ``Diviner`` model instance
     """
+    model_uri = str(model_uri)
+    # Determine if the file read indicator is present for saved in Spark or not
+    if RunsArtifactRepository.is_runs_uri(model_uri):
+        runs_uri = model_uri
+        model_uri = RunsArtifactRepository.get_underlying_uri(model_uri)
+        _logger.info("'%s' resolved as '%s'", runs_uri, model_uri)
+    elif ModelsArtifactRepository.is_models_uri(model_uri):
+        runs_uri = model_uri
+        model_uri = ModelsArtifactRepository.get_underlying_uri(model_uri)
+        _logger.info("'%s' resolved as '%s'", runs_uri, model_uri)
+
+    flavor_conf = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME)
 
     local_model_path = _download_artifact_from_uri(artifact_uri=model_uri, output_path=dst_path)
 
-    return _load_model(local_model_path)
+    _add_code_from_conf_to_system_path(local_model_path, flavor_conf)
 
+    if flavor_conf.get(_SPARK_MODEL_INDICATOR, False):
+        return _load_model_fit_in_spark(local_model_path, flavor_conf)
 
-def _get_diviner_instance_type(path) -> str:
```

Comment:
This is no longer used since we're loading the flavor configuration directly once within `load_model()`